rescript.opam

```diff
@@ -14,10 +14,11 @@ depends: [
   "cppo" {= "1.8.0"}
   "js_of_ocaml" {= "6.0.1"}
   "ounit2" {= "2.2.7"}
-  "dune"
+  "dune" {>= "3.17"}
+  "odoc" {with-doc}
```

Comment:
Why do we need to add odoc here, and why do we need the pinned -> dev change below?

Reply:
I don't know. The only change I made is the version in `opam-project` and the `wasm` option. Others are auto-generated by dune.

Reply:
Ah ok. CI is failing though, weird:

```
strip: unable to copy file '/home/runner/work/rescript/rescript/linuxarm64/rescript.exe'; reason: Permission denied
```

Reply:
![image](https://github.com/user-attachments/assets/1258b180-d3cb-427a-8e14-7f7ed32ea24a)

Oh it seems `dune build` now has extra checking and failed

Reply:
Seems the `[@@live]` doesn't work anymore

Reply:
Only on Windows is succeed lol

Reply:
Fixed.